src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -90,25 +90,81 @@ public Dictionary(int capacity, IEqualityComparer<TKey>? comparer)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.dictionary);
             }
 
+            if (dictionary.Count == 0)
```

Comment:
This introduces an additional interface call.  How about moving it into the if block below and doing the Count check off of `d` instead?  Or check to see whether the arrays are non-null, as they should only have been initialized by the base if the count was > 0?

Reply:
Could move to shared method and use for `Dictionary(IEnumerable<KeyValuePair<TKey, TValue>> collection)` also for when the `IEnumerable<KeyValuePair<TKey, TValue>>` is a Dictionary

Reply:
My preference would be to delete this check and the `if (newEntries.Length == length)` special case. It will leave some perf on the table in some situations, but I think it is the right tradeoff.

Reply:
Either need to do a `.Count == 0` check or a `_buckets is null` check as when the count is zero the arrays can be null

Reply:
Doing the check later once we know we are copying from Dictionary should be fine.